python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -38,22 +51,38 @@ class _ResolvableAssetAttributesMixin(BaseModel):
     automation_condition: Optional[str] = None
 
 
-class AssetAttributesSchema(_ResolvableAssetAttributesMixin, ComponentSchema):
+class AssetAttributesSchema(_ResolvableAssetAttributesMixin, ResolvableSchema[Mapping[str, Any]]):
```

Comment:
It is worth a comment here to explain why this returns a `Mapping[str, Any]` and what it represents